\section{Conclusion}
\label{sec:conclusion}

Because phylogenies are an abstraction that generalizes across all evolutionary processes, phylogenetic analysis can be applied across a breadth of biological and artificial life systems.
Consequently, a broad set of cross-disciplinary use cases exist for inferring the processes that shaped a phylogeny by quantifying its topology.
Indeed, extraction of information about evolutionary dynamics from phylogenetic history has been a longstanding and productive theme in evolutionary biology \citep{pagel1997inferring}.
This work seeks to contribute in that vein, by establishing foundations necessary to assess three fundamental evolutionary drivers --- spatial structure, ecology, and selection pressure --- in complex, distributed evolving populations.

First, we investigated the strength and character of structural signatures in phylogenetic histories left by selection pressure, ecology, and spatial population structure.
These drivers induced readily detectable effects across our suite of four surveyed phylometrics.
These effects were generally consistent across surveyed individual-level phylogenies, but differed notably under genotype-level tracking and under species-level simulation.
Although the directions of phylometric effects were generally consistent across treatments expected to increase phylogenetic richness (i.e., weak selection, ecologies, and spatial structure), effect significance/size for particular phylometrics differed among some treatments.
Compared to results from the simple explicit-fitness model, which was amenable to strong treatment manipulations, the more sophisticated Avida model system expressed fewer phylometric effects between treatments.

Compared to spatial structure and selection pressure, drivers designed to induce ecological coexistence exerted relatively muted, sparse effects on phylogenetic metrics.
However, follow-up experiments with both individual-level model systems revealed that phylometric signatures can remain detectable under background conditions of spatial population structure.
Interactions, though, are sometimes surprising.
A notable example is the reduced mean pairwise distance observed in Avida under conditions combining spatial structure and ecology, but not in the baseline ecology treatment.
These results highlight the complexity of how interacting evolutionary dynamics impact phylogenetic structure.
Even a single dynamic in isolation can influence phylometrics in opposite directions.
For instance, we found that changes to selection pressure in either direction significantly reduced trees' Colless-like index under the simple fitness-explicit model.

% TODO summarize specific examples as (i.e., metric for this vs. that)
Comparison of phylometrics from perfect-fidelity trees against corresponding reconstructions revealed that phylometric statistics differed in sensitivity to reconstruction error introduced by hereditary stratigraphy-based tracking.
Colless-like index and mean pairwise distance were more sensitive, and in some conditions, reconstruction persisted at even 1\% reconstruction resolution.
On the other hand, mean evolutionary distinctiveness and sum pairwise distance were particularly robust to reconstruction error, with no bias detected even at 33\% resolution reconstruction.
Results suggest 3\% reconstruction resolution as a reasonable ballpark parameterization for applications of hereditary stratigraphy with downstream phylogenetic analysis.
Where it did occur, the sign effect of phylometric bias was consistent, which might simplify considerations to account for it in experiments.

Our findings identify key next steps necessary for development of rigorous phylogenetic assays to test for phylogenetic influence of key evolutionary drivers.
Because treatments did not induce widespread opposite-sign effects, our results do not indicate a straightforward, qualitative approach to disambiguate drivers' signatures.
Future work should survey comprehensive panels of phylometrics, which might provide a stronger foothold to this end \citep{tuckerGuidePhylogeneticMetrics2017}.
Alternately, more sophisticated quantitative approaches leveraging differential effect size ratios may be fruitful.
Possibilities include application of machine learning techniques over broad phylometric panels \citep{voznica2022deep} or even inference directly over phylogenetic structure itself via graph neural networks and related methods \citep{lajaaiti2023comparison}.
The capability of artificial life approaches to generate high-quality data sets with high replication counts will be a significant asset to such work.
More information could be harnessed by incorporating trait data \citep{nozoe2017inferring} or by using time-sampled phylogenies \citep{volz2013viral}, which would include information about ``ghost'' lineages that have extincted at earlier time points.

Additionally, robust normalization methods will be crucial to enable meaningful comparisons between phylogenies differing in (1) population size, (2) population subsampling level, (3) depth of evolutionary history, and (4) demographic characteristics of organism life histories.
Despite some substantial work on this front \citep{shao1990tree,mir2018sound}, significant challenges remain.
It is worth noting that most analysis reported here normalizes to phylometric values sampled from ``plain'' evolutionary regimes, thereby implicitly depending on the availability of such data.
However, stripped-down baseline condition experiments may not always be possible, especially in biological model systems.
Extensive, likely sophisticated, considerations would be necessary to establish \textit{a priori} phylometric value predictions that could be compared against instead of testing against empirical values.
Finally, as evidenced in our experiments, reconstruction error can systematically bias phylometric values.
In scenarios where reconstruction error is unavoidable, additional normalizations may be necessary.
However, it is promising that some phylometrics appear thoroughly robust to reconstruction error and, in other cases, sufficient reconstruction accuracy might be achieved to obviate error bias.

The overriding objective of work presented here is to equip biologists and computational researchers with capability to investigate fundamental evolutionary drivers within large-scale evolving populations.
To this end, methodology must advance hand-in-hand with the software infrastructure required to put it into practice.
Artificial life experiments, specifically, tend to require particular instrumentation to collect phylogenetic histories, and we have been active in making general-use plug-and-play solutions available freely through the Python Packaging Index \citep{moreno2022hstrat,dolson2024phylotrackpy}.
As each field develops, artificial life will benefit greatly from domain-specific software and methods developed within traditional evolutionary bioinformatics and, we hope, vice versa.

% Together, our findings support development of more rigorous phylogenetic assays that might eventually be capable of identifying the evolutionary conditions that produced a phylogeny.
% look forward to continuing to push forward the

% Ultimately, the goal is to develop methodology that can be put into practice to increase the power of digital evolution and artificial life systems as experimental model systems.
% In addition to , this objective hinges on software availability.

% Production-ready software implementation available via the Python Packaging Index.
% C++ bindings for hereditary stratigraph genome annotation are under active development and will be released imminently.
% We are eager to collaborate to , and to contribute to developing large-scale (parallel and distributed) artificial life systems as an experimental tool



% Much future work

% \begin{itemize}
% \item better theory and methodology for null models; in this work we performed comparisons between replicates with different system parameterizations; in many contexts, this might not be practical or possible
% \item leverage these foundational observational findings to devise systematic testing protocols to identify evolutionary dynamics
% % \item incorporate information about ancestral state through ``fossils'' and see how well/accurately they can be tied into the tree of extant organisms (initial signs are good; unit tests with reconstructions from organisms sampled from different phylogenetic depths)
% % \item understanding complications of sexual recombinaton

% \end{itemize}

% Ultimately, the goal is to develop methodology that can be put into practice to increase the power of digital evolution and artificial life systems as experimental model systems.
% In addition to , this objective hinges on software availability.

% Production-ready software implementation available via the Python Packaging Index.
% C++ bindings for hereditary stratigraph genome annotation are under active development and will be released imminently.
% We are eager to collaborate to , and to contribute to developing large-scale (parallel and distributed) artificial life systems as an experimental tool

% recap work and findings
% In this paper, we focus on
% \begin{enumerate}
% \item need to be able to do fast, accurate tree reconstructions for very large distributed populations can efficiently create high-quality phylogenetic reconstructions for population sizes as large as 32,768
% \item need to understand the relationship between evolutionary dynamics and phylogenetic metrics
% \item
% \item nuisance factors of spatial structure and phylogenetic reconstruction error accompanying large-scale parallel and distributed digital evolution systems.

% \begin{enumerate}
% \item how much precision is required for accurate phylogenetic metrics?
% \item if bias is introduced, what type of bias (so could still detect "conservatively")
% \end{enumerate}
% \end{enumerate}
